### ISSUE TITLE
Tooltip disapearing with dataGrouping.

### ISSUE THREAD

**Issue by pawelfus** [highcharts/highcharts#1229]
Example: http://jsfiddle.net/TheEvilHammer/88eZG/2/

Click "All" in range selector - now hover one of the points, and wait until update will come. Tooltip disapears.
Now click on "1M" - hover of the points, wait untill update will come. Tooltip sticks to the hovered previously point (expected behaviour - as one of other issues on github).

**Comment by vickychijwani** [comment]
Hi, I'm facing this issue as well. I have a StockChart with dynamically-updated data and `dataGrouping` enabled. It's very annoying because we have updates coming in every 500 ms and the tooltip disappears instantly, which makes it useless. From debugging it appears that when new data comes in, the data groups are destroyed and created again, due to which the tooltip loses its `hoverPoints` and is hence hidden instead of being moved. Is there a fix / workaround for this? I can't just disable `dataGrouping` because I have 200,000-1,000,000 points. Thanks1

PS: the fiddle link is broken. Here's a new one, demoing the issue on HighStock v2.1.5: http://jsfiddle.net/g2paqg2a/

**Comment by highslide-software** [comment]
A full fix would require a refactoring of the data grouping logic, where we don't destroy the points after a new data grouping is performed, but instead tried to attach existing points to updated data. 

A quick workaround may be to override the tooltip.hide method so that it does nothing: http://jsfiddle.net/highcharts/g2paqg2a/1/

**Comment by vickychijwani** [comment]
Ok, thanks. I realize it's difficult to fix without refactoring a lot of stuff. Do you have any other suggestions for a better workaround? Never hiding the tooltip doesn't seem like a good idea to me.

**Comment by pawelfus** [comment]
How about displaying tooltip after adding point? `tooltip.refresh(point|points)` may be helpful. For example, store x-value for the last hovered point, and clear that value when mouse leave the container. Example: http://jsfiddle.net/g2paqg2a/3/

**Comment by highslide-software** [comment]
That's a nice hack @pawelfus!

**Comment by vickychijwani** [comment]
Thanks for these hacks! I got a chance to test @pawelfus's hack, and it almost works, except I don't know how to pass in the correct mouse event to `tooltip.refresh(points, mouseEvent)` (to make the tooltip stay in the same position). I looked at the event passed in to `plotOptions.series.point.events.mouseOver`, but it seems to be missing some essential properties (including `pageX` / `chartX`):

![mouseover-event](https://cloud.githubusercontent.com/assets/439616/7795244/a8b49c2c-02f0-11e5-90a7-309e3e9a0d2b.png)

What am I missing?
